src/aws-fsx/ontap-file-system.ts

```diff
@@ -0,0 +1,514 @@
+        automaticBackupRetentionDays: ontapConfiguration.automaticBackupRetention?.toDays() ?? 0,
+        dailyAutomaticBackupStartTime: ontapConfiguration.dailyAutomaticBackupStartTime?.toTimestamp(),
+        deploymentType: ontapConfiguration.deploymentType ?? OntapDeploymentType.MULTI_AZ_2,
+        diskIopsConfiguration: {
+          mode: ontapConfiguration.diskIops ? 'USER_PROVISIONED' : 'AUTOMATIC',
+          iops: ontapConfiguration.diskIops,
+        },
+        endpointIpAddressRange: ontapConfiguration.endpointIpAddressRange,
+        fsxAdminPassword: ontapConfiguration.fsxAdminPassword,
+        haPairs: ontapConfiguration.haPairs,
+        preferredSubnetId: ontapConfiguration.preferredSubnet?.subnetId,
+        routeTableIds: ontapConfiguration.routeTables?.map(routeTable => routeTable.routeTableId),
+        throughputCapacity: ontapConfiguration.throughputCapacity,
+        throughputCapacityPerHaPair: ontapConfiguration.throughputCapacityPerHaPair,
+        weeklyMaintenanceStartTime: ontapConfiguration.weeklyMaintenanceStartTime?.toTimestamp(),
+      },
+      securityGroupIds: [securityGroup.securityGroupId],
+      storageCapacity: props.storageCapacityGiB,
+    });
+
+    return fileSystem;
+  }
+
+  /**
+   * Validates the props provided for a new FSx for Ontap file system.
+   */
+  private validateProps(props: OntapFileSystemProps) {
+    const ontapConfiguration = props.ontapConfiguration;
+    const deploymentType = ontapConfiguration.deploymentType;
+
+    this.validateHaPairs(deploymentType, ontapConfiguration.haPairs);
+    this.validateAutomaticBackupRetention(ontapConfiguration.automaticBackupRetention);
+    this.validateDailyAutomaticBackupStartTime(
+      ontapConfiguration.automaticBackupRetention,
+      ontapConfiguration.dailyAutomaticBackupStartTime,
+    );
+    this.validateDiskIops(props.storageCapacityGiB, ontapConfiguration.diskIops, ontapConfiguration.haPairs);
+    this.validateEndpointIpAddressRange(deploymentType, ontapConfiguration.endpointIpAddressRange);
+    this.validateFsxAdminPassword(ontapConfiguration.fsxAdminPassword);
+    this.validateSubnets(deploymentType, props.vpcSubnets, ontapConfiguration.preferredSubnet);
+    this.validateRouteTables(deploymentType, ontapConfiguration.routeTables);
+    this.validateThroughputCapacity(
+      deploymentType,
+      ontapConfiguration.throughputCapacity,
+      ontapConfiguration.throughputCapacityPerHaPair,
+      ontapConfiguration.haPairs,
+    );
+    this.validateStorageCapacity(ontapConfiguration.haPairs, props.storageCapacityGiB);
+  }
+
+  private validateAutomaticBackupRetention(automaticBackupRetention?: Duration): void {
+    if (
+      automaticBackupRetention == null ||
+      automaticBackupRetention.isUnresolved() ||
+      automaticBackupRetention.toMilliseconds() === 0
+    ) {
+      return;
+    }
+    if (
+      automaticBackupRetention.toMilliseconds() < Duration.days(1).toMilliseconds() ||
```

Comment:
This validation is based on these description.

> AutomaticBackupRetentionDays
> The number of days to retain automatic backups. Setting this property to 0 disables automatic backups. You can retain automatic backups for a maximum of 90 days. The default is 30.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-fsx-filesystem-ontapconfiguration.html#cfn-fsx-filesystem-ontapconfiguration-automaticbackupretentiondays

Reply:
I've added that!